far/strmix.cpp

```diff
@@ -803,7 +803,7 @@ namespace
 	{
 		const auto BlankOrWordDiv = [&WordDiv](wchar_t Ch)
 		{
-			return std::iswblank(Ch) || contains(WordDiv, Ch);
+			return std::iswspace(Ch) || contains(WordDiv, Ch);
```

Comment:
As far as I remember, iswblank is used here because only space and \t were checked initially:

https://github.com/FarGroup/FarManager/blob/3f560a1d5f5b25f0795f92fef4d927fe839e2223/far/const.hpp#L49

Supposedly it was enough because \r, \n etc. are usually filtered out earlier.
iswspace looks reasonable if it is now used in Viewer, but it is probably worth checking other usages of iswblank just in case.

Reply:
First, I am confused by your link. I could not find this definition of `isspace(x)` on master.

Second, I looked quickly through the existing usages of `iswblank`. Most places seem appropriate; a few are less obvious. Audit of the latter requires either better knowledge of the codebase or much more time to understand the expected semantics. Given that all existing places now work with `iswblank`, it seems reasonable to suppose that the change is not required anywhere else. I strongly suggest leaving everything as is and wait for user's complaints.

So far, I renamed `GetSpaces` to `GetBlanks` and added all-new `GetSpaces` to make the naming consistent with the standard.

Reply:
> First, I am confused by your link. I could not find this definition of isspace(x) on master.

It is no longer there of course.
The link is just to explain why in many places we use iswblank and not iswspace, even though iswspace might appear more idiomatic in general.

> I strongly suggest leaving everything as is

I'm not suggesting changing anything in this PR, just making sure there are no other places that are accessed by Viewer::Search and potentially need an update.

And we probably should rename BlankOrWordDiv accordingly.

Reply:
Thanks for the explanation. As far as I can tell, `Viewer::Search` is clear. I renamed `BlankOrWordDiv` to `SpaceOrWordDiv`.

I also looked at all uses of `IsBlankOrEos` and `GetBlanks`. There are not many of them, and everywhere using "Space" instead of "Blank" may be reasonable, except for (at least in UCRT):
- `iswsapce` includes U+0085 (NEL) and U+00A0 (NBSP). It's an interesting question whether we want to break words at NBSP, even in Viewer Search. Maybe we should everywhere use `GetSpaces() + GetEols()` instead of `iswsapce`?
- For codepoints outside UNICODE C0/C1 (> U+00FF), `iswsapce` calls `GetStringTypeW` which returns who knows what. Again, switching to `GetSpaces() + GetEols()` may work more predictably.

What do you think?

Reply:
> It's an interesting question whether we want to break words at NBSP, even in Viewer Search

Why not. NBSP is about preventing line breaks, not word breaks.

Reply:
I often use NBSP in contexts like "64&nbsp;KiB" or "1776&nbsp;г." or "T.&nbsp;Pratchett". Not sure if "KiB" should be found in whole words mode. But then, why not? Let's keep `iswspace` for now.